src/Microsoft.Data/PrimitiveColumnComputations.cs

```diff
@@ -0,0 +1,1924 @@
+            var ret = (float)1;
+            for (int bb = 0 ; bb < column.Buffers.Count; bb++)
+            {
+                var buffer = column.Buffers[bb];
+                for (int i = 0; i < buffer.Length; i++)
+                {
+                    ret = (float)(buffer.Span[i] * ret);
+                    buffer.Span[i] = ret;
+                }
+            }
+        }
+
+        public void CumulativeSum(PrimitiveColumnContainer<float> column)
+        {
+            var ret = (float)0;
+            for (int bb = 0 ; bb < column.Buffers.Count; bb++)
+            {
+                var buffer = column.Buffers[bb];
+                for (int i = 0; i < buffer.Length; i++)
+                {
+                    ret = (float)(buffer.Span[i] + ret);
+                    buffer.Span[i] = ret;
+                }
+            }
+        }
+
+        public void Max(PrimitiveColumnContainer<float> column, out float ret)
+        {
+            ret = column.Buffers[0].Span[0];
+            for (int bb = 0 ; bb < column.Buffers.Count; bb++)
+            {
+                var buffer = column.Buffers[bb];
+                for (int i = 0; i < buffer.Length; i++)
+                {
+                    ret = (float)(Math.Max(buffer.Span[i], ret));
+                }
+            }
+        }
+
+        public void Min(PrimitiveColumnContainer<float> column, out float ret)
+        {
+            ret = column.Buffers[0].Span[0];
+            for (int bb = 0 ; bb < column.Buffers.Count; bb++)
+            {
+                var buffer = column.Buffers[bb];
+                for (int i = 0; i < buffer.Length; i++)
+                {
+                    ret = (float)(Math.Min(buffer.Span[i], ret));
+                }
+            }
+        }
+
+        public void Product(PrimitiveColumnContainer<float> column, out float ret)
+        {
+            ret = (float)1;
+            for (int bb = 0 ; bb < column.Buffers.Count; bb++)
+            {
+                var buffer = column.Buffers[bb];
+                for (int i = 0; i < buffer.Length; i++)
+                {
+                    ret = (float)(buffer.Span[i] * ret);
+                }
+            }
+        }
+
+        public void Sum(PrimitiveColumnContainer<float> column, out float ret)
+        {
+            ret = (float)0;
+            for (int bb = 0 ; bb < column.Buffers.Count; bb++)
+            {
+                var buffer = column.Buffers[bb];
+                for (int i = 0; i < buffer.Length; i++)
+                {
+                    ret = (float)(buffer.Span[i] + ret);
+                }
+            }
+        }
+
+        public void Round(PrimitiveColumnContainer<float> column)
+        {
+            for (int b = 0; b < column.Buffers.Count; b++)
+            {
+                var buffer = column.Buffers[b];
+                for (int i = 0; i < buffer.Length; i++)
+                {
+                    buffer.Span[i] = (float)(Math.Round((decimal)buffer.Span[i]));
```

Comment:
We should either be using MathF here, or casting to a `double` instead of decimal.

Reply:
Similar here. I wanted to keep the template file simple